File: MyPlanner.Data/Entities/Common/Page.cs
namespace MyPlanner.Data.Entities.Common;

public class Page : EntityBase
{
    public string Title { get; set; } = string.Empty;
    //Owner
    public string UserId { get; set; } = string.Empty;

    // Page to Page reference
    public List<Page> IncludePages { get; set; } = new List<Page>();
    public Guid? ParentPage { get; set; }

    // Content 
    public PageContent? Content { get; set; }
}

File: MyPlanner.Clients/React/src/entities/Pages/Page.ts
import PageContent from "./PageContent";

export default class Page {
    id: string;
    title: string;
    pages: Page[];
    content?: PageContent;
    constructor(id: string, title: string, pages: Page[] = [], content: PageContent | undefined = undefined) {
        this.id = id;
        this.title = title;
        this.pages = pages;
        this.content = content;
    }
}